### DISCUSSION TITLE
[ StockWhiz ] : An analysis environment for stocks, back testing, quantitative strategies, portfolio optimizations

### DISCUSSION THREAD

**Discussion by savakarrohan** [pymc-devs/pymc3#7280] [Show and tell]
[StockWhiz](https://www.stockwhiz.in) is a trading terminal coupled with a social media application. Aiming to provide traders the necessary tools to analyze and execute their trades.  

This is an appreciation post for pymc with a request for help, if StockWhiz falls in your cup of tea. 

https://github.com/pyodide/pyodide/assets/50200523/2070144a-a453-4d2f-8a18-e13dd5e3af67

YouTube Video : https://youtu.be/mMPHJnn8D9Y?si=CUeentT9dd1pwIOz

## Motivation
Backtesting and stock analysis is a crucial part of trading journey, in order to back test, one requires to have the respective packages installed, knowledge of coding and stock data. This is complex and a wide number of companies are attempting to address this problem by providing zero code interfaces to retail people. 

To make an effort in this direction I started with StockWhiz. Initially I used to conduct analysis for myself, and then I fell on [Pyodide](https://github.com/pyodide/pyodide) and [JupyterLite](https://github.com/jupyterlite/jupyterlite). I went ahead and built StockWhiz the pyodide backed backtesting engine. 

## Implementation 
StockWhiz has some salient features that are matched by only 3 other terminals worldwide. StockWhiz is built on Pyodide, The Jupyter ecosystem & Google's Blockly. The interface is simple and can be integrated on to any brokerage's charting library. It displays Buy/Sell signatures generated and important statistics for confidence of a trader. 

I think implementing Pyodide is the perfect solution for this problem of analysis of stocks. 
**Why?**
- The financial markets are places with massive amounts of data,
- python is ideal for quickly crunching data and providing analysis, 
- pyodide brings it at a single link. 

[StockWhiz](https://www.stockwhiz.in) does the exact same, a platform for stock analysis at the click of a single link. 

### Benefits
**Technical**
- Reduces cost by at least 4x (Current estimates, my gut says much much more.)
- Providing as a single package(file) to brokerages
- A complete analysis platform within 24mb

**Business**
- Unlimited backtests
    - All competitors run on a coupon-based system as they use cloud computing and thus have to balance pricing with features
- Breaking a monopoly
    - [TradingView](https://www.tradingview.com/) is the global leader for back testing, amongst others but they keep this part of their platform proprietary to themselves
- Scaling
    - Being able to provide it as a single JavaScript package allows integration with Brokerage platforms within a week, providing scale at ease.
    
#### Introducing myself
I @savakarrohan am a 24 year Bachelor's graduate of Mechanical from Indian Institute of Technology Dharwad and built StockWhiz from the ground up on top of open source packages. I have been devoted full time to StockWhiz since the past 6 months, while the coding journey started much earlier. 
You can reach out on [LinkedIn](https://www.linkedin.com/in/rohansavakar/) or email me at savakarrohan@stockwhiz.in

## Images 
![Screenshot 2024-02-14 155733](https://github.com/pyodide/pyodide/assets/50200523/81d84dda-477b-419d-b33a-257c27fba069)
<img width="979" alt="Screenshot 2023-12-21 174338" src="https://github.com/pyodide/pyodide/assets/50200523/2d5bf234-5c88-4a1b-b69a-c56025947666">
![Screenshot 2024-03-30 185413](https://github.com/pyodide/pyodide/assets/50200523/d9558b08-cbef-455b-ba99-f63667fb559c)
![Screenshot 2024-03-26 223109](https://github.com/pyodide/pyodide/assets/50200523/cbcc1452-5ade-4c13-93b6-00dcbffba2af)

**Comment by ricardoV94** [comment]
Hi @savakarrohan,

Thanks for sharing. I missed what is the help request?
